src/main/java/org/wso2/sample/identity/oauth2/grant/ed25519/X509GrantHandler.java

```diff
@@ -126,10 +139,11 @@ private String getUserFromSSHCert(String sshCertificate) {
 
                 if (publicKey instanceof Certificate) {
                     Certificate certificate = (Certificate) publicKey;
-
                     List<String> validPrincipals = certificate.getValidPrincipals();
+
                     if (validPrincipals != null && !validPrincipals.isEmpty()) {
                         String principal = validPrincipals.get(0);
```

Comment:
```suggestion
                        // Only the 1st user principal is considered
                        String principal = validPrincipals.get(0);
```

Reply:
Resolved in https://github.com/gayashanbc/x509-ed25519-grant-handler/pull/3/commits/d01c3a2846a13850ced1a2e93b47bb46687eb8e1